src/Frontend/Fragment/Comment.hs

```diff
@@ -22,10 +22,9 @@ import qualified Frontend.Path as U
 
 
 data CommentWidget = CommentWidget
-    { _cwRenderContext :: RenderContext
-    , _cwIdeaCaps      :: [Capability]
-    , _cwComment       :: Comment
-    , _cwPhase         :: Phase
+    { _cwCapCtx   :: CapCtx
+    , _cwIdeaCaps :: [Capability]
```

Comment:
So, the `CapCtx` which was used to create the `cwIdeaCaps` was less precise than `cwCapCtx`. The one in `cwCapCtx` has a `Just comment`.

So if we were to reuse the caps coming out of `cwCapCtx` for idea-level conditions then there wouldn't be redundant computations either. However we would rely on the monotonic property of `capabilities`. Namely if a `c0 c1 : CapCtx` where `c1` is more defined than `c0` then `capabilities c0` should be a subset of `capabilities c1`.

If we agree on this analysis then we can throw away `cwIdeaCaps`.